### ISSUE TITLE
Bart model for textinfilling

### ISSUE THREAD

**Issue by andompesta** [huggingface/transformers#4735]
# ❓ Questions & Help

## Details

@sshleifer
I was wandering why you used a registered_buffer to define the ``final_logits_bias`` in the BartForConditionalGeneration. In my understanding a registered_buffer would have not gradient so if I fine-tune the a Bart model for generation these biases would not be updated (remain to 0). Should we use  register_parameter or remove the bias completely or there is something I haven't understand ? 

**A link to original question on Stack Overflow**:

**Comment by sshleifer** [comment]
it is a hack to make `MarianMTModel`, which inherits from Bart work. For the Bart models that parameter does nothing, as you suggest. You can remove or ignore.

**Comment by andompesta** [comment]
Thanks got it
